metadata/metadata.go

```diff
@@ -300,3 +300,34 @@ func (c *Client) WriteGuestAttributes(ctx context.Context, key, value string) er
 	_, err = c.httpClient.Do(req)
 	return err
 }
+
+// GetKey gets a specific metadata key.
+func (c *Client) GetKey(key string) (string, error) {
+	req, err := http.NewRequest("GET", c.metadataURL+key, nil)
+	if err != nil {
+		return "", err
+	}
+	req.Header.Add("Metadata-Flavor", "Google")
+
+	res, err := c.httpClient.Do(req)
+	if err == nil {
```

Comment:
Is it really returning nil error in case of 404/412 error code, or do you mean to have `err != nil` condition?

Reply:
Nope, I indeed meant `err == nil`,  in 404 and 412 error will be nil but with StatusCode set.

Reply:
Got it, this should also check for non-nil error then before attempting to read response

Reply:
Sure. Will get it fixed. Thanks.

Reply:
Done.